code/modules/research/ordnance/_scipaper.dm

```diff
@@ -0,0 +1,312 @@
+ * If you want to subtype this, do it in a way that doesn't mess with the type change.
+ */
+/datum/scientific_paper/proc/clone(typepath)
+	var/datum/scientific_paper/new_paper = new typepath
+	new_paper.title = title
+	new_paper.author = author
+	new_paper.et_alia = et_alia
+	new_paper.abstract = abstract
+	new_paper.gains = gains
+	new_paper.experiment_path = experiment_path
+	new_paper.tracked_variable = tracked_variable
+	new_paper.tier = tier
+	new_paper.partner_path = partner_path
+	return new_paper
+
+/// Returns the formatted, readable gist of our paper in a list.
+/datum/scientific_paper/proc/return_gist()
+	var/list/gist = list()
+	var/list/transcripted_gains = list(SCIPAPER_COOPERATION_INDEX, SCIPAPER_FUNDING_INDEX)
+	for (var/index in 1 to transcripted_gains.len)
+		if (!gains)
+			transcripted_gains[index] = "None"
+		switch (round(gains[index]))
+			if(-INFINITY to 0)
+				transcripted_gains[index] = "None"
+			if(1 to 24)
+				transcripted_gains[index] = "Little"
+			if(25 to 49)
+				transcripted_gains[index] = "Moderate"
+			if(50 to 99)
+				transcripted_gains[index] = "Significant"
+			if(100 to INFINITY)
+				transcripted_gains[index] = "Huge"
+			else
+				transcripted_gains[index] = "Undefined"
+	gist["gains"] = transcripted_gains
+	gist["title"] = title
+	gist["author"] = author
+	gist["etAlia"] = et_alia
+	gist["abstract"] = abstract
+	gist["experimentName"] = initial(experiment_path?.name)
+	gist["tier"] = tier
+	gist["partner"] = initial(partner_path?.name)
+	return gist
+
+/datum/scientific_paper/proc/autofill()
+	if(!title)
+		title = "On [initial(experiment_path.name)] - [tier]"
+	if(!author)
+		author = "Unknown"
+		et_alia = FALSE
+	if(!abstract)
+		abstract = "Published on [station_time_timestamp()]"
+
+/datum/scientific_paper/explosive
+	/** 
+	 * Used to check explosive experiments that needs to be unique.
+	 * References a tachyon record where applicable.
+	*/
+	var/datum/data/tachyon_record/explosion_record
+
+/// Check if our explosion has already been published and whether the experiment path is correct or not.
+/datum/scientific_paper/explosive/allowed_to_publish(datum/techweb/techweb_to_check)
+	if(!ispath(experiment_path, /datum/experiment/ordnance/explosive))
+		return FALSE
+	if(!istype(explosion_record))
+		return FALSE
+	var/list/published_papers = techweb_to_check.published_papers
+	for(var/experiment_path in published_papers)
+		for(var/datum/scientific_paper/explosive/papers in published_papers[experiment_path])
+			var/datum/data/tachyon_record/record_to_check = papers.explosion_record
+			if(explosion_record.explosion_identifier == record_to_check.explosion_identifier)
+				return FALSE
+	. = ..()
+
+/datum/scientific_paper/explosive/set_experiment(ex_path = null, variable = null, data = null)
+	if(!ispath(ex_path, /datum/experiment/ordnance/explosive) || !variable || !istype(data, /datum/data/tachyon_record))
+		for (var/index in 1 to args.len)
+			args[index] = null
```

Comment:
excuse me? the fuck is this

Reply:
I want each override of the proc to handle only the vars initialized at their own subtype, and then leave their parent to handle the rest. That code means we found something sus on our level, thus we need to tell the parent that the paper will be invalidated.

Reply:
That seems like a really messy pattern, why do that?

Reply:
I dont remember why, probably got bored of normal inheritance. I'll get rid of it.